src/app/shared/services/http.service.ts

```diff
@@ -50,8 +50,8 @@ export class HttpService {
     filterHeader: string[],
     metadataNames: string[],
     storage: string,
-  ): Observable<Report[]> {
-    return this.http.get<Report[]>('api/metadata/' + storage + '/', {
+  ): Observable<Record<string, string>[]> {
+    return this.http.get<Record<string, string>[]>(`api/metadata/${storage}/`, {
```

Comment:
It was because there were functions that were used in retrieveRecords that required it to be of type Record<string, string> so it could be looped over. I fixed this so it's still being looped over while being of type Report

Reply:
This function is still Record<string, string> though?